etna/datasets/tsdataset.py

```diff
@@ -1039,16 +1040,33 @@ def get_level_dataset(self, target_level: str) -> "TSDataset":
                 target_level=target_level, source_level=self.current_df_level
             )
 
-            source_level_data = self[:, current_level_segments, "target"].values
+            column_names = self.columns.get_level_values(level=1)
+            target_names = tuple(match_target_quantiles(column_names))
+            target_names = target_names + ("target",)
+
+            num_target_names = len(target_names)
+            num_current_level_segments = len(current_level_segments)
+            num_target_level_segments = len(target_level_segments)
+
+            source_level_data = self[:, current_level_segments, target_names].values
+
+            source_level_data = source_level_data.reshape((-1, num_current_level_segments, num_target_names))
+            source_level_data = np.swapaxes(source_level_data, 1, 2)
+            source_level_data = source_level_data.reshape((-1, num_current_level_segments))
```

Comment:
Do we actually need this reshape, broadcasting should work here

Reply:
Can you explain this idea in more detail ?